pkgs/development/libraries/physics/fastnlo_toolkit/default.nix

```diff
@@ -15,17 +16,23 @@
 
 stdenv.mkDerivation rec {
   pname = "fastnlo_toolkit";
-  version = "2.3.1pre-2411";
+  rev = "2823";
```

Comment:
This declares a bash variable `$rev`. I don't suppose the build process needs that? Please move it into `passthru` if you need it, or remove it.

Is it a tag or a short commit hash? If it's a tag, please rename it to `tag`. If it's a short commit hash, please make it longer, so the fetcher won't run into an ambiguous commit hash error as upstream development continues.

Reply:
@roberth @SuperSandro2000 There needs to be some limit to nitpicking. We could set it to "if you would not make a PR to fix this, don't bring this up". In this case, the `fetchGit*` doesn't support `tag`, it supports `rev`, I did not invent the name. There are also plenty derivations that rely on `rec` instead of `let rev=""; in`, how about fixing those.

Reply:
So we're just dumping every variable into the sandbox from now on?

The things you write are supposed to be meaningful, not necessarily as short or convenient as can be. If other packages fail to do so, that is also a problem, not an excuse.

Another non-goal is minimizing repetition in string literals, specifically because it's not meaningful when the pattern is incidental, so I agree with you @veprbl regarding the other comment.

Reply:
I was genuinely confused by the value of `rev` because it looks more like a short commit than a tag.

I understand your frustration. I don't feel strongly about this anymore since I've checked that git "commit-like" resolution behaves in a way that won't break the fetcher, but I would prefer not having to look that up.

Reply:
> So we're just dumping every variable into the sandbox from now on?

You imply that this PR sets a precedent, but `grep -rB3 "rev =" . | grep -A3 mkDerivation` disagrees with you.

> I was genuinely confused by the value of `rev` because it looks more like a short commit than a tag.

From the history, my guess is that the project used to run on svn, but not anymore. Yet they still have numerical git tags similar to svn revision id, I don't know how they calculate those, but they are still in use (for now?). I could rename it to `svn_rev`, that can be confusing in its own way.